content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md

```diff
@@ -278,10 +278,11 @@ manager, and scheduler run as [DaemonSet pods](/docs/concepts/workloads/controll
 configured via the Kubernetes API instead of [static pods](/docs/tasks/administer-cluster/static-pod/)
 configured in the kubelet via static files.
 
+To create a self-hosted cluster, pass the flag `--feature-gates=SelfHosting=true` to `kubeadm init`.
+
 {{< caution >}}
-**Caution:** Self-hosting is alpha, but is expected to become the default in
-a future version. To create a self-hosted cluster, pass the `--feature-gates=SelfHosting=true`
-flag to `kubeadm init`.
+**Caution:** `SelfHosting` is an alpha feature. It is deprecated in 1.12
```

Comment:
Should we let users know what solution will replace the current feature? If some users depend on this feature, they might want to know for now what they can do in the future.

Reply:
@xiangpengzhao 

@Bradamant3 just requested that we remove the hint that this feature might be replaced eventually:
https://github.com/kubernetes/website/pull/10348#discussion_r219015410

also, this is quite unclear ATM and still in the planning stage:
https://github.com/kubernetes/kubeadm/issues/1093

probably best to not link to that issue for now, too.

Reply:
Didn't notice that comment. SGTM.
/lgtm